app/src/main/java/de/westnordost/streetcomplete/quests/railway_crossing/AddRailwayCrossingBarrierForm.kt

```diff
@@ -4,18 +4,21 @@ import de.westnordost.streetcomplete.R
 import de.westnordost.streetcomplete.quests.AImageListQuestAnswerFragment
 import de.westnordost.streetcomplete.view.Item
 
-class AddRailwayCrossingBarrierForm : AImageListQuestAnswerFragment<String,String>() {
+class AddRailwayCrossingBarrierForm : AImageListQuestAnswerFragment<AddRailwayCrossingBarrierForm.BarrierOption,RailwayCrossingBarrierAnswer>() {
+    override fun onClickOk(selectedItems: List<BarrierOption>) {
+        if(selectedItems.lastIndex == 0) {
+            applyAnswer(RailwayCrossingBarrierAnswer(selectedItems[0]))
+        }
```

Comment:
That is part of what `List.single()` does in Kotlin. It 1. asserts that there is only one item in the list, and 2. returns it.